tests/backward-compat/yahoo-custom-version/src/test/groovy/org/apache/bookkeeper/tests/backwardcompat/TestCompatUpgradeYahooCustom.groovy

```diff
@@ -170,11 +170,16 @@ class TestCompatUpgradeYahooCustom {
             openAndVerifyEntries(currentCL, currentBK, ledger4.getId())
             assertCantWrite(yahooCL, ledger4)
 
-            // yahoo client can fence a bookie created by current client
+            // Since METADATA_VERSION is upgraded and it is using binary format, the older
+            // clients which are expecting text format would fail to read ledger metadata.
             def ledger5 = createAndWrite(currentCL, currentBK)
             ledgers.add(ledger5.getId())
-            openAndVerifyEntries(yahooCL, yahooBK, ledger5.getId())
-            assertCantWrite(currentCL, ledger5)
+            try {
+                openAndVerifyEntries(yahooCL, yahooBK, ledger5.getId())
+            } catch (Exception exc) {
+                Assert.assertEquals(exc.getClass().getName(),
```

Comment:
Do we still need this test?
This is telling now that we are not compatible by default

Reply:
I would prefer to keep this. I would see it as more like with this change it is expected to fail with the specific exception. Also, it would serve as record (history) that this change had broke backward compatibility.